scanpy/plotting/_tools/scatterplots.py

```diff
@@ -1066,15 +1078,16 @@ def _basis2name(basis):
 def _process_image(adata, data_points, img_key, crop_coord, scale_spot, bw=False):
     offset = 100
     cmap_img = None
-    img = adata.uns['images'][img_key]
+    library_id = list(adata.uns['spatial'].keys())[0]
+    img = adata.uns['spatial'][library_id]['images'][img_key]
```

Comment:
```suggestion
    spatial_data = adata.uns["spatial"][library_id]
    img = spatial_data['images'][img_key]
```

And then the same for the other long paths below too.

Reply:
Done